caravel/model/moderation.py

```diff
@@ -0,0 +1,46 @@
+from google.appengine.ext import ndb
+from caravel.model import principal
+
+class ModeratedMixin(ndb.Model):
+    TYPE_ONCE_APPROVED = None
+
+    def _pre_put_hook(self):
```

Comment:
Er, good question. The docstring is functionally correct — but we need to do some munging in order to update the Key's type name.